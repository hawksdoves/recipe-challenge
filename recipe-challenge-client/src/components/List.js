import React from 'react';
import { Link } from 'react-router-dom';
import { Card } from 'semantic-ui-react';

import Recipe from './Recipe';
import Favourite from './Favourite';

function List(props) {
    const { recipes, favourites, ...rest } = props;

    return recipes.length > 0 ?
        (<Card.Group >
            { recipes.map(recipe => {
                return (  
                    <div key={recipe.id} className='recipe' >
                        <Link to={`/recipe/${recipe.id}`} key={`Link-${recipe.id}`} className='card'>
                            <Recipe recipe={recipe} />
                        </Link>
                        <Favourite favourite={favourites.includes(recipe.id)} {...rest} recipeId={recipe.id} />
                    </div>
                )
            })}
        </Card.Group> ) :
        <div>Sorry, nothing matched your filter term</div>
};

export default List;
