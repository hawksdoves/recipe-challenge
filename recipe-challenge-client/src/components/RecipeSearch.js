import React from 'react';
import { Icon, Input } from 'semantic-ui-react';

function RecipeSearch(props) {
    const { searchRecipes } = props;

    return <Input 
                className='recipeSearch' 
                type="text" 
                placeholder="Recipe Search..." 
                onChange={searchRecipes} 
                icon={<Icon name='search' inverted circular link />} 
            />
}

export default RecipeSearch;
