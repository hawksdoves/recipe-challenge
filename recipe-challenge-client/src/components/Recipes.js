import React from 'react';

import RecipeSearch from './RecipeSearch';
import List from './List';

class Recipes extends React.Component {

    searchRecipes = (event) => {
        const updatedResults = this.props.recipes.filter(recipe => {
            return recipe.name.toLowerCase().search(
                event.target.value.toLowerCase()) !== -1;
            });
        this.setState({ recipeResults: updatedResults});
    }

    state = {
        recipeResults: null
    }

    render() {
        const { recipes, ...rest } = this.props

        return  recipes && recipes.length > 0 ? (
            <div className='recipesContainer'>
                <RecipeSearch searchRecipes={this.searchRecipes} />
                <List recipes={this.state.recipeResults || recipes} {...rest} />
            </div>
        ) :
            <div>Sorry, we currently have no recipes for you</div>
    }
}

export default Recipes;
