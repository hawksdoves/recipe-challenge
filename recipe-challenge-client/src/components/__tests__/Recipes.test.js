import React from 'react';
import { shallow, mount } from 'enzyme';

import RecipeSearch from '../RecipeSearch'
import Recipes from '../Recipes';
import List from '../List';

const recipes = [
    {
        id: '1',
        name: 'steak'         
    },
    {
        id: '2',
        name: 'chicken'
    }
];

describe('Recipes', () => {
    it('should render RecipeSearch and List', () =>{
        const favourites = [];
        const wrapper = shallow(<Recipes recipes={recipes} />);

        expect(wrapper).toMatchSnapshot();
    })

    it('should display message if there are no recipes', () =>{
        const favourites = [];
        const wrapper = shallow(<Recipes recipes={[]} favourites={favourites} />);

        expect(wrapper).toMatchSnapshot();
    })

    it('should update recipeResults state when searchRecipes is called', () =>{
        const favourites = [];
        const wrapper = shallow(<Recipes recipes={recipes} favourites={favourites} />);
        const searchRecipesSpy = jest.spyOn(wrapper.instance(), 'searchRecipes');
        const setStateSpy = jest.spyOn(wrapper.instance(), 'setState');
        const event = { target: { value: 'steak' } };
        wrapper.instance().searchRecipes(event);

        expect(searchRecipesSpy).toHaveBeenCalledWith(event);
        expect(setStateSpy).toHaveBeenCalled();
    })

});
