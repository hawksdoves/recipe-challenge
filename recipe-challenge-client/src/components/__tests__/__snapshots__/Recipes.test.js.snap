// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recipes should display message if there are no recipes 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    Sorry, we currently have no recipes for you
</div>,
  "nodes": Array [
    <div>
      Sorry, we currently have no recipes for you
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Recipes
        favourites={Array []}
        recipes={Array []}
/>,
      "_debugID": 3,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Recipes {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "favourites": Array [],
          "recipes": Array [],
        },
        "refs": Object {},
        "searchRecipes": [Function],
        "state": Object {
          "recipeResults": null,
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 2,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          Sorry, we currently have no recipes for you
</div>,
        "_debugID": 4,
        "_renderedOutput": <div>
          Sorry, we currently have no recipes for you
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Recipes
    favourites={Array []}
    recipes={Array []}
/>,
}
`;

exports[`Recipes should render RecipeSearch and List 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="recipesContainer"
>
    <RecipeSearch
        searchRecipes={[Function]}
    />
    <List
        recipes={
            Array [
                Object {
                  "id": "1",
                  "name": "steak",
                },
                Object {
                  "id": "2",
                  "name": "chicken",
                },
              ]
        }
    />
</div>,
  "nodes": Array [
    <div
      className="recipesContainer"
>
      <RecipeSearch
            searchRecipes={[Function]}
      />
      <List
            recipes={
                  Array [
                        Object {
                          "id": "1",
                          "name": "steak",
                        },
                        Object {
                          "id": "2",
                          "name": "chicken",
                        },
                      ]
            }
      />
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Recipes
        recipes={
                Array [
                        Object {
                          "id": "1",
                          "name": "steak",
                        },
                        Object {
                          "id": "2",
                          "name": "chicken",
                        },
                      ]
        }
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Recipes {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "recipes": Array [
            Object {
              "id": "1",
              "name": "steak",
            },
            Object {
              "id": "2",
              "name": "chicken",
            },
          ],
        },
        "refs": Object {},
        "searchRecipes": [Function],
        "state": Object {
          "recipeResults": null,
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="recipesContainer"
>
          <RecipeSearch
                    searchRecipes={[Function]}
          />
          <List
                    recipes={
                              Array [
                                        Object {
                                          "id": "1",
                                          "name": "steak",
                                        },
                                        Object {
                                          "id": "2",
                                          "name": "chicken",
                                        },
                                      ]
                    }
          />
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="recipesContainer"
>
          <RecipeSearch
                    searchRecipes={[Function]}
          />
          <List
                    recipes={
                              Array [
                                        Object {
                                          "id": "1",
                                          "name": "steak",
                                        },
                                        Object {
                                          "id": "2",
                                          "name": "chicken",
                                        },
                                      ]
                    }
          />
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Recipes
    recipes={
        Array [
            Object {
              "id": "1",
              "name": "steak",
            },
            Object {
              "id": "2",
              "name": "chicken",
            },
          ]
    }
/>,
}
`;
